### PR TITLE
[TS] LPS-172025 Document Preview in Workflow shows metadata inconsistently

### PR DESCRIPTION
**Pull request by ricardocousolr**
Hi @liferay-workflow team,

Documents under a workflow are shown extra information about only when there's no preview available (see JSP https://github.com/liferay-workflow/liferay-portal/blob/master/modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/asset/file_entry_full_content.jsp#L34). 

Since the availability of this information shouldn't depend on this, we can align all cases by using the boolean `showExtraInfo` that will make sure that information is always shown if `true`. 

Please let me know if you have any questions. 

Thanks!

### CODE REVIEW

**Comment by naraandrade** [comment]
ci:test:workflow

**Comment by ricardocousolr** [comment]
Hi team!

Are the failures related to the solution? Is there anything I can do to move the revision along? 

Thanks!

File: modules/apps/portal-workflow/portal-workflow-task-web/src/main/resources/META-INF/resources/view_content.jsp [hunk 1]
```diff
@@ -65,6 +65,7 @@ renderResponse.setTitle(assetRenderer.getTitle(workflowTaskDisplayContext.getTas
 						assetEntry="<%= assetEntry %>"
 						assetRenderer="<%= assetRenderer %>"
 						assetRendererFactory="<%= assetRendererFactory %>"
+						showExtraInfo="<%= true %>"
```
Comment (carolmariaabb) [inline modules/apps/portal-workflow/portal-workflow-task-web/src/main/resources/META-INF/resources/view_content.jsp:68]: @ricardocousolr I'm a little concerned that this change might affect other use cases.. For example, other places that are expecting `false`..
Is it possible to make this more specific, like [here](https://github.com/liferay/liferay-portal/blob/045dfcd7cf0f5630008e30b11e4694570adea96a/modules/apps/asset/asset-publisher-web/src/main/java/com/liferay/asset/publisher/web/internal/display/context/AssetPublisherDisplayContext.java#L1942-L1951)? In our case, you could create a new method on `WorkflowTaskDisplayContext.java` where the default value is `false` and D&M would be responsible for changing this value.

Comment (ricardocousolr) [reply modules/apps/portal-workflow/portal-workflow-task-web/src/main/resources/META-INF/resources/view_content.jsp:68]: Thanks for your review @carolmariaabb! I'll implement those changes and update the pr.

Comment (ricardocousolr) [reply modules/apps/portal-workflow/portal-workflow-task-web/src/main/resources/META-INF/resources/view_content.jsp]: Hi again @carolmariaabb, 

I've pushed a new commit ([9e4549a](https://github.com/liferay-workflow/liferay-portal/pull/872/commits/9e4549aee7cd1e8c711c0275364526b5c936a508)) with the changes. Please let me know if I that was what you meant or if I should do any changes. 

Thanks!

File: modules/apps/portal-workflow/portal-workflow-task-web/src/main/java/com/liferay/portal/workflow/task/web/internal/display/context/WorkflowTaskDisplayContext.java [hunk 2]
```diff
@@ -685,6 +685,17 @@ public boolean isShowEditURL(WorkflowTask workflowTask) {
 		return showEditURL;
 	}
 
+	public boolean isShowExtraInfo() {
+		if (_showExtraInfo != null) {
+			return _showExtraInfo;
+		}
+
+		_showExtraInfo = ParamUtil.getBoolean(
+			_httpServletRequest, "showExtraInfo", true);
```
Comment (carolmariaabb) [inline modules/apps/portal-workflow/portal-workflow-task-web/src/main/java/com/liferay/portal/workflow/task/web/internal/display/context/WorkflowTaskDisplayContext.java:694]: @ricardocousolr what happens if we don't set the `defaultValue` to `getBoolean`? I mean if we remove the `true` and do `httpServletRequest.setAttribute("showExtraInfo", Boolean.TRUE);` when necessary.

Comment (ricardocousolr) [reply modules/apps/portal-workflow/portal-workflow-task-web/src/main/java/com/liferay/portal/workflow/task/web/internal/display/context/WorkflowTaskDisplayContext.java:694]: @carolmariaabb, could you elaborate on your alternative approach?

Comment (carolmariaabb) [reply modules/apps/portal-workflow/portal-workflow-task-web/src/main/java/com/liferay/portal/workflow/task/web/internal/display/context/WorkflowTaskDisplayContext.java:694]: Hi @ricardocousolr,

After investigating the code a little more, the fix that seems to be the most appropriate would look like this [code](https://github.com/liferay-continuous-integration/liferay-portal/blob/34c002de7efce150a2d69a3749938054a41bec49/modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/result/display/builder/SearchResultContentDisplayBuilder.java#L114-L115), i.e.:

```
public boolean isShowExtraInfo() {
	if (_showExtraInfo != null) {
		return _showExtraInfo;
	}

	if (Objects.equals(
			ParamUtil.getString(_liferayPortletRequest, "type"),
			"document")) {

		_showExtraInfo = true;
	}

	return _showExtraInfo;
}
```
Makes sense to you?

Comment (ricardocousolr) [reply modules/apps/portal-workflow/portal-workflow-task-web/src/main/java/com/liferay/portal/workflow/task/web/internal/display/context/WorkflowTaskDisplayContext.java]: Hi @carolmariaabb, 

yes I think targeting documents like that is the most conservative approach. 

I've push a couple of commits to the PR:
- [c3b6ecb](https://github.com/liferay-workflow/liferay-portal/pull/872/commits/c3b6ecb821ab78f6b8ef4bbbe8f81a187427575e) uses your change but making sure that `_showExtraInfo` is set to `fasle` if previously `null` and not of `"document"` type.
- [5e2bd09](https://github.com/liferay-workflow/liferay-portal/pull/872/commits/5e2bd09be15966d63dcbcc83beb0e849cf7280cf) applies the same approach to another display context and view that I hand't noticed existed. These can be accessed from:
1. Personal Menu Icon > My Submissions (`UserWorkflowPortlet`)
2. Upper Menu > Applications > Submissions (`ControlPanelWorkflowInstancePortlet`)

Let me know if you think this is on the right track. Thanks!

**Review by carolmariaabb — APPROVED** [review]
LGTM!

**Comment by carolmariaabb** [comment]
ci:test:relevant

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:forward

**Pull request closed without merging**